xhttp/fake_client_test.go

```diff
@@ -0,0 +1,70 @@
+package xhttp_test
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+)
+
+// FakeClient allows to fake interactions with an [xhttp.Client].
+type FakeClient struct {
+	requests  []*http.Request
+	responses []response
+	mutex     sync.Mutex
+}
+
+func NewFakeClient() *FakeClient {
```

Comment:
Code that injects the xhttp.Client interface could use this FakeClient to test interactions with other services (investiganting requests + injecting responses).

Reply:
If this makes sense we can just publish it with the package :-)

Reply:
I think that's a good idea. Users of xhttp might want to use it in their own tests.